tests/cli/test_cli.py

```diff
@@ -0,0 +1,169 @@
+from sqlmesh.cli.main import cli
+
+
+@pytest.fixture(scope="session")
+def runner() -> CliRunner:
+    return CliRunner()
+
+
+def create_example_project(temp_dir) -> None:
+    """
+    Sets up CLI tests requiring a real SQLMesh project by:
+        - Creating the SQLMesh example project in the temp_dir directory
+        - Overwriting the config.yaml file so the duckdb database file will be created in the temp_dir directory
+    """
+    init_example_project(temp_dir, "duckdb")
+    with open(temp_dir / "config.yaml", "w") as f:
+        f.write(
+            f"""gateways:
+    local:
+        connection:
+            type: duckdb
+            database: {temp_dir}/db.db
+
+default_gateway: local
+
+model_defaults:
+    dialect: duckdb
+"""
+        )
+
+
+def reset_example_project(temp_dir) -> None:
+    rmtree(temp_dir / ".cache")
+    os.unlink(temp_dir / "db.db")
+
+
+def assert_duckdb_test(result) -> None:
+    assert "Successfully Ran 1 tests against duckdb" in result.output
+
+
+def assert_new_env(result, new_env="prod", from_env="prod") -> None:
+    assert f"New environment `{new_env}` will be created from `{from_env}`" in result.output
+
+
+def assert_backfill_success(result) -> None:
+    assert "All model versions have been created successfully" in result.output
+    assert "All model batches have been executed successfully" in result.output
+    assert "The target environment has been updated successfully" in result.output
+
+
+def assert_plan_success(result, new_env="prod", from_env="prod") -> None:
+    assert result.exit_code == 0
+    assert_duckdb_test(result)
+    assert_new_env(result, new_env, from_env)
+    assert_backfill_success(result)
+
+
+def test_plan(runner, tmp_path):
+    # Error if no SQLMesh project config is found
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan"])
+    assert result.exit_code == 1
+    assert "Error: SQLMesh project config could not be found" in result.output
+
+    create_example_project(tmp_path)
+
+    # Example project models have start dates, so there are no date prompts
+    # for the `prod` environment. User input `y` is required to apply the plan.
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan"], input="y\n")
+    assert_plan_success(result)
+    reset_example_project(tmp_path)
+
+    # Example project plan for `prod` runs end-to-end with no user input if `--auto-apply` is passed
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan", "--auto-apply"])
+    assert_plan_success(result)
+    reset_example_project(tmp_path)
+
+    # Successful test run message should not appear if `--skip-tests` is passed and plan is applied
+    result = runner.invoke(cli, ["--paths", tmp_path, "plan", "--skip-tests"], input="y\n")
+    assert result.exit_code == 0
+    assert "Successfully Ran 1 tests against duckdb" not in result.output
+    assert_new_env(result)
+    assert_backfill_success(result)
+
+    rmtree(tmp_path)
```

Comment:
This shouldn't be needed. `tmp_path` is unique per test so each test is running in it's own isolated space and therefore you can let pytest deal with the cleanup. It actually can be nice to leave the old path around for debugging purposes. I think pytest clears out past tests once they are over 5 test runs old.